presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -158,10 +159,10 @@ private static ConnectorPageSource createParquetPageSource(
         try {
             FileSystem fileSystem = hdfsEnvironment.getFileSystem(user, path, configuration);
             FileStatus fileStatus = fileSystem.getFileStatus(path);
-            long fileSize = fileStatus.getLen();
+            long estimatedFileSize = fileStatus.getLen();
```

Comment:
This is an exception because actually the getFileStatus call here will actually be the exact file size now. If this code changes to avoid the status call, then it'll need to handle start >= actual file size and start + length > actual file size just like the other page sources.

Reply:
I believe the same reasoning about split region applies to Iceberg.